src/main/scala/org/hatdex/hat/api/models/User.scala

```diff
@@ -58,7 +58,13 @@ object UserRole {
           case "applicationmanage"        => ApplicationManage(extra)
           case "applicationstatusread"    => ApplicationStatusRead(extra)
           case "managefiles"              => ManageFiles(extra)
-          case "emailverified"            => EmailVerified()
+          // Use Verified as a generic indicator of verification.
+          // The `extra` field can differentiate the type of verification
+          // These "stack" as well, so a user can have multiple Verified roles.
+          case "verifiedviaemail"         => Verified("email")
```

Comment:
Here, we need to have a single "verified" case that converts into `Verified(extra)` case class to match the `...extends UserRole("verified")` implementation on line 118

Reply:
ok